ip: Remove dependency on ipnet crate

cathay4t opened the pull request:
The new ipnet 2.8 release breaks our compiling. Considering we only use
a very few functions from ipnet, I have replace the ipnet usage to self
maintaining code.

Unit test cases included.

maiqueb commented on rust/src/lib/ip.rs:
```diff
@@ -1245,23 +1245,73 @@ fn get_ip_prop_list(
+            ErrorKind::InvalidArgument,
+            format!(
+                "Invalid IP network string: '{ip_net}', expecting \
+                    192.0.2.0/24 or 2001:db8:1::/64 or 192.0.2.1"
```
are these examples ?

cathay4t replied:
Yes

ffmancera replied:
I think using these addresses are a little bit confusing. It could indicate the user to use these addresses. IMO, we should refer them to documentation about IPv4/IPv6 address formatting or make it clear that these are examples.

cathay4t replied:
Changed to

```
                "Invalid IP network string: '{ip_net}', expecting 'ip/prefix' \
                or 'ip' format, for example: 192.0.2.0/24 or \
                2001:db8:1::/64 or 192.0.2.1"
```

maiqueb replied:
perfect, looks better IMO. 

Thanks @ffmancera for making the concern I had more explicit :)

maiqueb commented on rust/src/lib/unit_tests/ip.rs:
would we want to unit test scenario with net prefix length > maximum one ?

Just asking because your code actually checks for that.

cathay4t replied:
Unit test cases added to cover all the errors raised by production code.

maiqueb reviewed (approved):
Pretty far from a rust expert but the code looks good.

Just a short comment about the coverage of the unit tests.

cathay4t committed:
ip: Remove dependency on ipnet crate

The new ipnet 2.8 release breaks our compiling. Considering we only use
a very few functions from ipnet, I have replace the ipnet usage to self
maintaining code.

Unit test cases included.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t commented:
New patch uploaded with unit test coverage expanded.
